content/articles/2019-08-lifecycle-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,255 @@
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-soft-deprecated.svg")
+```
+
+A __soft-deprecated__ feature can still be used without hassle, but users should consider switching to an alternative approach.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-deprecated.svg")
+```
+
+A __deprecated__ feature is likely to be discontinued in the next major release. Users should switch to an alternative approach as soon as possible.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-defunct.svg")
+```
+
+A __defunct__ function is still exported, and a defunct argument is still part of the signature. However, the feature can no longer be used, it now throws an informative error.
+
+
+### Adding lifecycle badges in documentation
+
+You can insert the lifecycle badges anywhere in your documentation. First import the badges in your package:
+
+```{r, eval = FALSE}
+# At the time of writing, you need the development version of usethis
+remotes::install_github("r-lib/usethis")
+usethis::use_lifecycle()
+```
+
+Then use the Rd macro `\lifecycle{stage}`:
+
+```{r, eval = FALSE}
+#' \lifecycle{experimental}
+#' \lifecycle{soft-deprecated}
+```
+
+You typically don't need to advertise the status of a function if it is the same as the package as a whole. For instance, if your package is [maturing](https://www.tidyverse.org/lifecycle/#maturing), only signal functions in the experimental, stable, questioning stages, or deprecated stages. A good place to include the lifecycle badge of a function is at the top of the `@description` block. If the function is deprecated, we recommend marking it as internal so it no longer appear in the help index:
+
+```{r, eval = FALSE}
+#' Recoup some local heads outside the sent staged changes
+#'
+#' @description
+#'
+#' \lifecycle{soft-deprecated}
+#'
+#' `git_recoup_head()` recoups any non-indexed downstream heads before a
+#' few relinked local archives.
+#'
+#' @keywords internal
+```
+
+The lifecycle stage of your function is now clearly indicated in its documentation topic:
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/recoup-deprecated.png")
+```
+
+To document the lifecycle stage of an argument, insert the badge in the argument description:
+
+```{r, eval = FALSE}
+#' @param indict_branch \lifecycle{deprecated}
+#'
+#'   If `FALSE`, `git_blend_tag()` parses references from
+#'   the filter-branched upstream.
+```
+
+```{r, echo = FALSE}
+knitr::include_graphics("/images/lifecycle/indict-deprecated.png")
+```
+
+
+### Verbosity of deprecation
+
+As is convention in R, usage of deprecated functions is signalled with warnings. However, lifecycle tries to warn in a non-invasive way. When a deprecated function is called multiple times, only the first time causes a warning. The function then works silently for the rest of the session.
```

Comment:
```suggestion
Deprecated functions use warnings to signal that you need to update your code. lifecycle tries to do this in a non-invasive way, as too many warnings are almost as annoying as an error. To avoid overwhelming you with warnings, lifecycle only warns the first time a deprecated function is used.
```